misc/python/materialize/cli/bazel.py

```diff
@@ -168,17 +173,59 @@ def output_path(target) -> list[pathlib.Path]:
     return [pathlib.Path(path) for path in paths]
 
 
-def remote_cache_arg() -> list[str]:
+def remote_cache_arg(config: Config) -> list[str]:
     """List of arguments that could possibly enable use of a remote cache."""
 
-    # TODO(parkmycar): Setup access to the remote cache for developers with Teleport.
     ci_remote = os.getenv("CI_BAZEL_REMOTE_CACHE")
+    config_remote = config.bazel.remote_cache
 
     if ci_remote:
-        return [f"--remote_cache={ci_remote}"]
+        remote_cache = ci_remote
+    elif config_remote:
+        bazel_remote = RemoteCache(config_remote)
+        remote_cache = bazel_remote.address()
+    else:
+        remote_cache = None
+
+    if remote_cache:
+        return [f"--remote_cache={remote_cache}"]
     else:
         return []
 
 
+class RemoteCache:
+    """The remote cache we're conecting to."""
+
+    def __init__(self, value: str):
+        if value.startswith("teleport"):
+            app_name = value.split(":")[1]
+            self.kind = RemoteCacheKind.teleport
+            self.data = app_name
+        else:
+            self.kind = RemoteCacheKind.normal
+            self.data = value
+
+    def address(self) -> str:
+        """Address for connecting to this remote cache."""
+        if self.kind == RemoteCacheKind.normal:
+            return self.data
+        else:
+            TeleportProxy.spawn(self.data, "8080")
+            return "http://localhost:8080"
```

Comment:
8080 seems awfully likely to conflict with other local HTTP servers folks might be running. Can we pick something more obscure (idk, like 6889)?

Reply:
Good call, updated!